packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -76,19 +76,30 @@ export interface CheckoutEvents {
 	/**
 	 * {@inheritdoc TreeViewEvents.commitApplied}
 	 */
-	commitApplied(data: CommitMetadata, getRevertible?: () => Revertible): void;
-
-	/**
-	 * Fired when a revertible is either reverted or discarded.
-	 *
-	 * This event can be used to maintain a list or set of active revertibles.
-	 * @param revertible - The revertible that was disposed.
-	 * This revertible was previously passed to the `newRevertible` event.
-	 * Calling `discard` on this revertible is not necessary but is safe to do.
-	 */
-	revertibleDisposed(revertible: Revertible): void;
+	commitApplied(data: CommitMetadata, getRevertible?: RevertibleFactory): void;
 }
 
+/**
+ * Factory for creating a {@link Revertible}.
```

Comment:
I think you mentioned offline that it is bad to call this more than once. Is that true? Should the factory fail if you call it twice - and should we document that here?